### ISSUE TITLE
Add support for Line Break Best Practices - Two or more spaces

### ISSUE THREAD

**Issue by Thomas-Haase** [kovetskiy/mark#61]
When uploading Markdown files that use **two (or more) spaces at end of line to create a line break**, as recommended e.g. at [Line Break Best Practices](https://www.markdownguide.org/basic-syntax/#line-breaks), this results in **Confluence pages** that **have no line break** at this position. In fact even no whitespace at all is generated at this position, i.e. the lines are concatenated without any whitespace.
My test content was:
```txt
This is the first line.  
And this is the second line. 
```
This was rendered as:
```txt
This is the first line.And this is the second line.
```
I have tried alternatively the following scenarios, which both work as expected:
- When inserting the same Markup content into the Confluence 'Insert markup' window, the result is rendered as expected.
- When using `<br \>` instead of the two spaces, the result is as expected after uploading them with *mark*.

**Comment by ingydotnet** [comment]
The most important thing here is that consecutive text lines are joined without a space.
```
This is
the first
sentence.
And this
is the second
sentence.
```

is rendered as:
```
This isthe firstsentence.And thisis the secondsentence.
```

This pretty much makes the `mark` unusable for a ton of Markdown, at the moment.

**Comment by kovetskiy** [comment]
@ingydotnet @Thomas-Haase @chrisdavidmiles I've identified the issue, it is a regression after #53. I'll fix it soon. You can roll back to mark v4.0 to render your markdowns for now.

**Comment by ingydotnet** [comment]
@kovetskiy thanks for responding so quick!

I'll roll it back for now.

**Comment by kovetskiy** [comment]
@ingydotnet @Thomas-Haase would be great if you guys can try #c22c6c1e438c6777c764dfc277626d7bd58f8aa4 and let me know if it works or not before I release it. I added the tests this time (the lack of tests was the root cause of all the problems actually), but still would be great to know it works on real documents.

**Comment by Thomas-Haase** [comment]
Yes of course I would be happy to check this with my real documents. I can easily remove the sed workaround in may workflow, which replaces the two spaces with the HTML-ish break.
But I have no Go compiler available. I would need a binary for  linux x64 to test

**Comment by ingydotnet** [comment]
@kovetskiy I can confirm that c22c6c1 works for me. Well a coworker tested that (for commit c22c6c1 and others):
```
foo
bar
```
renders to
```
<p>foo
bar</p>
```
same as 4.0, rather than the 4.1
```
<p>foobar</p>
```

**Comment by whizard** [comment]
I did test it against c22c6c1 and it did the "correct" thing. When will 4.3 be released? Thanks for a great utility!

**Comment by kovetskiy** [comment]
Just released it as 5.0 since it uses the next version of the markdown compiler — [blackfriday](https://github.com/russross/blackfriday).

Docker image is also available.

**Comment by Thomas-Haase** [comment]
@kovetskiy I confirm, that 5.0 fixes my issue report. You are very fast, really great, excellent tool, many thanks. 
Unfortunatley I noticed an new regression in 5.0 over 4.4 in a completely different context. First I will check if my content is somehow incorrect or insane, then I will open a separate issue
